plugins/flashrom/fu-flashrom-lspcon-i2c-spi-device.c

```diff
@@ -117,19 +188,29 @@ fu_flashrom_lspcon_i2c_spi_device_set_version (FuDevice *device, GError **error)
 	g_autofree gchar *contents = NULL;
 	g_autofree gchar *version = NULL;
 	struct flashrom_flashctx *flashctx = NULL;
+	g_autoptr(FlashromLayout) layout = create_flash_layout ();
 	gsize flash_size;
 	guint32 addr;
+	gint active_partition;
 
 	/* set up flashrom layout */
 	flashctx = fu_flashrom_device_get_flashctx (flashrom_device);
-	flashrom_layout_set (flashctx, &layout);
+	flashrom_layout_set (flashctx, layout);
+
+	/* get the active partition */
+	if ((active_partition = probe_active_flash_partition (self, error)) == -1)
+		return FALSE;
```

Comment:
ew, can probe_active_flash_partition please return gboolean and then just pass active_partition by reference as the second parameter.

Reply:
That does make rather more sense.